app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java

```diff
@@ -0,0 +1,36 @@
+package org.schabi.newpipe.settings;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.preference.Preference;
+
+import org.schabi.newpipe.CheckForNewAppVersionTask;
+import org.schabi.newpipe.R;
+
+public class UpdateSettingsFragment extends BasePreferenceFragment {
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        String updateToggleKey = getString(R.string.update_app_key);
+        findPreference(updateToggleKey).setOnPreferenceChangeListener(updatePreferenceChange);
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        addPreferencesFromResource(R.xml.update_settings);
+    }
+
+    private Preference.OnPreferenceChangeListener updatePreferenceChange
+            = new Preference.OnPreferenceChangeListener() {
```

Comment:
You can use a lambda here.

Reply:
Hey, I'm not very familiar with lambda. Could you please suggest how to do I implement that? Or any resource which can?

Reply:
Instead of writing:
```
setListener(new Listener() {
   void theEvent(int parameter) {
      ...
   }
```
In java 7+ you can simply write:
```
setListener( (int parameter) -> {
   ...
}
```

Its shorter and simpler to read.